src/main/java/org/embulk/input/marketo/rest/MarketoRestClient.java

```diff
@@ -0,0 +1,335 @@
+            }
+        }
+    }
+
+    public InputStream getLeadBulkExtractResult(String exportId)
+    {
+        return getBulkExtractResult(MarketoRESTEndpoint.GET_LEAD_EXPORT_RESULT, exportId);
+    }
+
+    public InputStream getActivitiesBulkExtractResult(String exportId)
+    {
+        return getBulkExtractResult(MarketoRESTEndpoint.GET_ACTIVITY_EXPORT_RESULT, exportId);
+    }
+
+    private InputStream getBulkExtractResult(MarketoRESTEndpoint endpoint, String exportId)
+    {
+        InputStream fileStream = doGet(this.endPoint + endpoint.getEndpoint(new ImmutableMap.Builder().put("export_id", exportId).build()), null, null, new MarketoFileResponseEntityReader(READ_TIMEOUT_MILLIS));
+        return fileStream;
+    }
+
+    public RecordPagingIterable<ObjectNode> getLists()
+    {
+        return getRecordWithPagination(endPoint + MarketoRESTEndpoint.GET_LISTS.getEndpoint(), null, ObjectNode.class);
+    }
+
+    public RecordPagingIterable<ObjectNode> getPrograms()
+    {
+        return getRecordWithPagination(endPoint + MarketoRESTEndpoint.GET_PROGRAMS.getEndpoint(), null, ObjectNode.class);
+    }
+
+    public RecordPagingIterable<ObjectNode> getLeadsByProgram(String programId, List<String> fieldNames)
+    {
+        Multimap<String, String> multimap = ArrayListMultimap.create();
+        multimap.put("fields", StringUtils.join(fieldNames, ","));
+        return getRecordWithPagination(endPoint + MarketoRESTEndpoint.GET_LEADS_BY_PROGRAM.getEndpoint(new ImmutableMap.Builder().put("program_id", programId).build()), multimap, ObjectNode.class);
+    }
+
+    public RecordPagingIterable<ObjectNode> getLeadsByList(String listId, List<String> fieldNames)
+    {
+        Multimap<String, String> multimap = ArrayListMultimap.create();
+        multimap.put("fields", StringUtils.join(fieldNames, ","));
+        return getRecordWithPagination(endPoint + MarketoRESTEndpoint.GET_LEADS_BY_LIST.getEndpoint(new ImmutableMap.Builder().put("list_id", listId).build()), multimap, ObjectNode.class);
+    }
+
+    public RecordPagingIterable<ObjectNode> getCampaign()
+    {
+        return getRecordWithPagination(endPoint + MarketoRESTEndpoint.GET_CAMPAIGN.getEndpoint(), null, ObjectNode.class);
+    }
+
+    private <T> RecordPagingIterable<T> getRecordWithPagination(final String endPoint, final Multimap<String, String> parameters, final Class<T> recordClass)
+    {
+        return new RecordPagingIterable<>(new RecordPagingIterable.PagingFunction<RecordPagingIterable.MarketoPage<T>>()
+        {
+            @Override
+            public RecordPagingIterable.MarketoPage<T> getNextPage(RecordPagingIterable.MarketoPage<T> currentPage)
+            {
+                final Multimap<String, String> params = ArrayListMultimap.create();
+                params.put(BATCH_SIZE, batchSize + "");
+                params.put(NEXT_PAGE_TOKEN, currentPage.getNextPageToken());
```

Comment:
Seems to be code duplication from here and the one below as the only difference is `params.put(NEXT_PAGE_TOKEN, currentPage.getNextPageToken());`.

Reply:
Fixed